src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -960,39 +965,38 @@ Error MeshForwarder::GetFragmentPriority(Lowpan::FragmentHeader &aFragmentHeader
     return error;
 }
 
-void MeshForwarder::GetForwardFramePriority(const FrameData      &aFrameData,
-                                            const Mac::Addresses &aMeshAddrs,
-                                            Message::Priority    &aPriority)
+void MeshForwarder::GetForwardFramePriority(const RxInfo &aRxInfo, Message::Priority &aPriority)
 {
     Error                  error      = kErrorNone;
-    FrameData              frameData  = aFrameData;
+    RxInfo                 rxInfo     = aRxInfo;
```

Comment:
nit: we only need to copy `mFrameData`, right? I can see that copying `RxInfo` is simpler to understand but also results in copying more data. Not sure which is better. Thoughts?

Reply:
We need a copy of `RxInfo` for the `GetFramePriority()` call below.

An alternative would be to create an overload of `GetFramePriority()` that accepts the parameters separately. I feel it might be simpler to do a copy here. Currently, `RxInfo` contains `FrameData` and the `MacAddrs` (approximately ~26 bytes), which should be small enough to copy without significant overhead.

A side goal for adding `RxInfo` is to later track additional info. Specifically, I noticed that we may decompress and parse IPv6 headers multiple times. I want to later include  `Ip6::Headers` within `RxInfo`, and track whether we've already parsed the IPv6 headers. This could then make copying `RxInfo` more expensive. To address this, my idea to then add a custom `operator=()` overload to copy only specific fields.